### PR TITLE
Use Test Proxy in azure-storage-queue

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Updates `azure-storage-queue` to use Test Proxy instead of the in-process `PlaybackHttpClient`. As part of this change `azure-storage-queue` was moved from using Groovy/Spock for testing to JUnit 5.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/BuildHelperTests.java [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class BuildHelperTests {
+    private static final ClientLogger LOGGER = new ClientLogger(BuilderHelper.class);
```
Comment (samvaity) [inline sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/BuildHelperTests.java:54]: BuildHelperTests.class

File: sdk/storage/azure-storage-queue/src/test/resources/session-records/AzuriteTests.azuriteUrlConstructingQueueClientWithConnectionString[1].json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (samvaity) [inline sdk/storage/azure-storage-queue/src/test/resources/session-records/AzuriteTests.azuriteUrlConstructingQueueClientWithConnectionString[1].json:1]: Should these be marked @DoNotRecord?

Comment (alzimmermsft) [reply sdk/storage/azure-storage-queue/src/test/resources/session-records/AzuriteTests.azuriteUrlConstructingQueueClientWithConnectionString[1].json:1]: I'm going to leave that for now as this is just a one-to-one port

**Review by ibrahimrabab — APPROVED** [review]

**Pull request merged**